.github/workflows/publish-maven-snapshots.yml

```diff
@@ -0,0 +1,34 @@
+name: Publish snapshots to maven
+
+on:
+  workflow_dispatch:
+  push:
+      branches:
+        - main
+        - 1.x
+
+jobs:
+  build-and-publish-snapshots:
+    runs-on: ubuntu-latest
+
+    permissions:
+      id-token: write
+      contents: write
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          distribution: adopt
+          java-version: 11
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.PUBLISH_SNAPSHOTS_ROLE }}
+          aws-region: us-east-1
+
+      - name: Publish snapshots to maven
+        run: |
+          ./gradlew --no-daemon publishPublishMavenPublicationToSnapshotRepoRepository -PORG_GRADLE_PROJECT_snapshotRepoUsername=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-username --query SecretString --output text) -PORG_GRADLE_PROJECT_snapshotRepoPassword=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-password --query SecretString --output text)
```

Comment:
Hi @reta I have not tried this but I feel this is much safer than environment variables. Let me know what you think,

Reply:
I have not tried this either, do you see the risks that args like `${aws ... )` could leak raw secrets into logs?

Reply:
Keeping it what works for everyone for now. Can you please review again? Thanks!